modules/manifold/src/components/ui/segment-groups-control/__tests__/ui-segment-panels.spec.js

```diff
@@ -15,6 +15,10 @@ test('utils: updateSegmentGroups', () => {
   expect(updateSegmentGroups([[0, 1, 2], [3]], 0, 3)).toEqual([[0, 1, 2], [3]]);
   // no action if segment 3 is the only segment in group 1 when deselecting from segment 1
   expect(updateSegmentGroups([[0, 1, 2], [3]], 1, 3)).toEqual([[0, 1, 2], [3]]);
+  // add a segment that originally belonged to neither group
+  expect(updateSegmentGroups([[0, 1], [3]], 0, 2)).toEqual([[0, 1, 2], [3]]);
+  // no action if updating will cause one of the groups to become empty
+  expect(updateSegmentGroups([[0, 1], [2]], 0, 2)).toEqual([[0, 1], [2]]);
```

Comment:
Not in this PR but let's use an object as the input to these functions for better dev experience.
The current parameter signature makes it difficult to infer what does `[0, 1]`, `[2]`, `0`, and `2` mean.

Reply:
Added todo.